ntoskrnl/ps/job.c

```diff
@@ -707,6 +708,60 @@ NtQueryInformationJobObject (
 
             break;
 
+        case JobObjectBasicProcessIdList:
+
+            /* Lock */
+            KeEnterGuardedRegionThread(CurrentThread);
+            ExAcquireResourceSharedLite(&Job->JobLock, TRUE);
+
+            NeededSize = sizeof(JOBOBJECT_BASIC_PROCESS_ID_LIST) - RTL_SIZEOF_THROUGH_FIELD(JOBOBJECT_BASIC_PROCESS_ID_LIST, ProcessIdList) +
+                Job->ActiveProcesses * RTL_SIZEOF_THROUGH_FIELD(JOBOBJECT_BASIC_PROCESS_ID_LIST, ProcessIdList);
+
+            /* We'll handle this */
+            NoOutput = TRUE;
+
+            if (JobInformationLength >= NeededSize)
+            {
+                ULONG Total = 0;
+
+                _SEH2_TRY
+                {
+                    BasicProcessIdList = JobInformation;
+                    BasicProcessIdList->NumberOfProcessIdsInList = Job->ActiveProcesses;
+
+                    Status = STATUS_SUCCESS;
+
+                    for (NextEntry = Job->ProcessListHead.Flink;
+                        NextEntry != &Job->ProcessListHead;
+                        NextEntry = NextEntry->Flink)
+                    {
+                        if (Total == Job->ActiveProcesses)
+                        {
+                            DPRINT1("ERROR: Job->ActiveProcesses did not match Job->ProcessListHead!");
+                            Status = STATUS_PROCESS_IN_JOB;
```

Comment:
Not sure what to return here.

Reply:
I do not think you're actually supposed to return a status code in this kind of situation. Whenever a process' unique ID is not associated within a job, `NumberOfAssignedProcesses` field is decremented and the operation moves on until you have looped thorough all the present processes in the job.

With that said, `NumberOfAssignedProcesses` has to be set to active processes and when a process' ID is indeed associated with that job, you increment `NumberOfProcessIdsInList` and submit the process ID to the list.

Reply:
This `ActiveProcesses` count is relied upon earlier to ensure the buffer was big enough.
If there is a mismatch between this number and the LIST_ENTRY there is probably a serious problem?
Might even warrant an assert.